src/gui/application.cpp

```diff
@@ -665,7 +682,14 @@ void Application::parseOptions(const QStringList &options)
             }
         }
         else {
-            showHint("Unrecognized option '" + option.toStdString() + "'");
+            QString errorMessage;
+            if (!AccountSetupCommandLineManager::instance()->parseCommandlineOption(option, it, errorMessage)) {
+                if (!errorMessage.isEmpty()) {
+                    showHint(errorMessage.toStdString());
+                    return;
+                }
+                showHint("Unrecognized option '" + option.toStdString() + "'");
+            }
```

Comment:
Are we missing a destroy here?

Reply:
@claucambra Not really. We don't want to destroy it until all the arguments are parsed and we either have started creating an account or proceed with a normal launch. We do the destroy after parsing of all arguments is finished and we have triggered or not triggered an account creation job. Otherwise, we have lost the data we parsed really, as the instance got destroyed.

Reply:
Okay, thanks for the explainer